AutomatedTesting/Gem/PythonTests/atom_renderer/atom_hydra_scripts/hydra_AtomEditorComponents_DepthOfFieldAdded.py

```diff
@@ -3,139 +3,149 @@
     import azlmbr.legacy.general as general
     import azlmbr.math as math
-    import azlmbr.paths
 
-    from editor_python_test_tools import hydra_editor_utils as hydra
-    from editor_python_test_tools.utils import Report
-    from editor_python_test_tools.utils import TestHelper as helper
-    from editor_python_test_tools.utils import Tracer
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper as helper
 
     with Tracer() as error_tracer:
-        # Wait for Editor idle loop before executing Python hydra scripts.
+        # Test setup begins.
+        # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
         helper.init_idle()
+        helper.open_level("Physics", "Base")
 
-        # Delete all existing entities initially for test setup.
-        search_filter = azlmbr.entity.SearchFilter()
-        all_entities = entity.SearchBus(azlmbr.bus.Broadcast, "SearchEntities", search_filter)
-        editor.ToolsApplicationRequestBus(bus.Broadcast, "DeleteEntities", all_entities)
-
-        # Set up camera entity and component.
-        camera_entity = hydra.Entity("camera_entity")
-        camera_entity.create_entity(math.Vector3(512.0, 512.0, 34.0), ["Camera"])
-        Report.result(Tests.camera_creation, camera_entity.id.IsValid())
-
-        # 1. Creation of entity with DepthOfField component.
+        # Test steps begin.
+        # 1. Creation of DepthOfField entity.
         depth_of_field = "DepthOfField"
-        depth_of_field_entity = hydra.Entity(f"{depth_of_field}")
-        depth_of_field_entity.create_entity(math.Vector3(512.0, 512.0, 34.0), [depth_of_field])
-        has_component = hydra.has_components(depth_of_field_entity.id, [depth_of_field])
-        Report.result(Tests.depth_of_field_creation, depth_of_field_entity.id.IsValid())
-        Report.critical_result(Tests.depth_of_field_component, has_component)
-
-        # 2. UNDO the entity creation.
-        general.undo()
-        Report.result(Tests.creation_undo, depth_of_field_entity.id.IsValid())
-
-        # 3. REDO the entity creation.
-        general.redo()
-        Report.result(Tests.creation_redo, depth_of_field_entity.id.isValid())
-
-        # 4. Enter/Exit game mode.
+        depth_of_field_entity = EditorEntity.create_editor_entity_at(
+            math.Vector3(512.0, 512.0, 34.0), f"{depth_of_field}")
+        Report.critical_result(Tests.depth_of_field_creation, depth_of_field_entity.exists())
+
+        # 2. Add a DepthOfField component to DepthOfField entity.
+        depth_of_field_component = depth_of_field_entity.add_component(depth_of_field)
+        Report.critical_result(Tests.depth_of_field_component, depth_of_field_entity.has_component(depth_of_field))
+
+        # 3. UNDO the entity creation and component addition.
+        # Requires 3 UNDO calls to remove the Entity completely.
```

Comment:
i would rather make the comments as loose as possible to make maintenance and future updates easier
the multiple UNDO's are to match the multiple actions so that it can actually UNDO it for the `exists()` check - which part is a defect? we can cut a bug for that but im not clear on the defect part i guess

Reply:
the comment just sounds like it works in a defective way pointing out that you have to undo 3 times for this to remove the entity. the code above just adds an entity and adds a component which sounds like 2 steps in history...
you can ignore this comment